### PR TITLE
add a resolver and validate command

### PR DESCRIPTION
**Pull request by vutran**
- Rename constants to match its label
- New `Resolver` class - validates a list of repositories (partially addresses #7)
- New command `zel validate` which validates a local `.zel` file
  - Flag: `--failFast` - terminate process on errors
- Adds bluebird (partially addresses #1, used on `Resolver`)

### CODE REVIEW

File: lib/local.js [hunk 1]
```diff
@@ -30,10 +30,9 @@ const readLocalFile = () => new Promise((resolve, reject) => {
  * @return {Promise<String[]>} - The list of local dependencies
  */
 const getLocalDependencies = () => new Promise((resolve, reject) => {
-    readLocalFile().then((file) => {
-        const deps = file.dependencies;
-        return deps ? resolve(deps) : reject('No local dependencies defined. Please define a repository.');
-    }).catch(err => reject(err));
+    readLocalFile()
+        .then(files => resolve(files.dependencies || []))
```
Comment (lukeed) [inline lib/local.js]: You're not `rejecting` here because of the `validate`, right? Would cancel early

File: index.js [hunk 2]
```diff
@@ -4,29 +4,54 @@ const prog = require('caporal');
...
     downloadRepo(repo).then((files) => {
-        files.forEach(file => logger.info(OK, `${file} ${FROM} ${repo}`));
+        files.forEach(file => logger.info(DOWNLOADED, `${file} ${FROM} ${repo}`));
     }).catch(err => logger.error(ERROR, err));
 }
 
 function initLocal(logger) {
     getLocalDependencies().then((deps) => {
+        if (!deps.length) {
+            logger.error(ERROR, 'No local dependencies defined. Please define a repository.');
```
Comment (lukeed) [inline index.js]: Could also `throw` and let the parent `catch` handle it. Redefining same handler.

Comment (vutran) [inline index.js]: Ah, good `catch`!

File: lib/resolver.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+        this.repositories.forEach((repoName) => {
+            this.validateSingle(repoName)
+                .then((repoName) => this.emit('valid', repoName))
+                .catch((repoName) => this.emit('invalid', repoName));
+        });
+    }
+
+    /**
+     * Validates a single repository by fetching it's dotfile
```
Comment (lukeed) [inline lib/resolver.js]: its

File: index.js (hunk 2 above)
Comment (lukeed) [inline index.js]: Hah 😆

**Comment by lukeed** [comment]
Looks good 👍

File: index.js [hunk 4]
```diff
@@ -4,29 +4,54 @@ const prog = require('caporal');
...
     .argument('[query]', 'Specify the repository to fetch.')
     .action((args, options, logger) => {
         if (args.query) {
             return init(args.query, logger);
         }
 
         return initLocal(logger);
+    })
+
+    .command('validate', 'Validates the .zel file.')
```
Comment (lukeed) [inline index.js]: `Validates local .zel file` ?

File: lib/local.js (hunk 1 above)
Comment (vutran) [inline lib/local.js]: I believe so. I totally forgot why I did this and realized `files.dependencies` doesn't make sense here. 😩

Comment (lukeed) [inline lib/local.js]: Oh haha, missed that. You meant `file` and `file.dependencies`

Comment (lukeed) [inline lib/local.js]: But yeah, it makes sense. In the (odd) event a user was validating a local `.zel` file that didn't include any dependencies, the old `reject` would interrupt the `validate` action.

File: index.js [hunk 5]
```diff
@@ -8,19 +8,22 @@ const Resolver = require('./lib/resolver');
 const { FROM, ERROR, DOWNLOADED, VALID, INVALID } = require('./lib/constants');
 
 function init(repo, logger) {
-    downloadRepo(repo).then((files) => {
-        files.forEach(file => logger.info(DOWNLOADED, `${file} ${FROM} ${repo}`));
-    }).catch(err => logger.error(ERROR, err));
+    downloadRepo(repo)
+        .then((files) => {
```
Comment (lukeed) [inline index.js]: I'm still figuring out your code style 😉

Comment (vutran) [inline index.js]: Hah! I like chain functions on new lines, though forget to do so sometimes.

**Comment by lukeed** [comment]
Should I update your eslint rules? It's yelling at me about parentheses around single argument.

**Comment by vutran** [comment]
> Should I update your eslint rules? It's yelling at me about parentheses around single argument.

Yeah. I really dislike airbnb's config due to this inconsistency.

**Pull request merged**